### PR TITLE
Deprecate Nullable methods

### PR DESCRIPTION
**Pull request by ararslan**
If nothing else, the (temporary) dependency on Nullables should fix the errors on 0.7. (Edit: Well, some of them, anyway...)

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -7,7 +7,8 @@ module TestCategoricalArrays
     quiet = length(ARGS) > 0 && ARGS[1] == "-q"
     anyerrors = false
 
-    using Base.Test
+    using Compat
+using Compat.Test
```
Comment (nalimilan) [inline test/runtests.jl]: Indentation bug.

File: src/deprecated.jl [hunk 2]
```diff
@@ -9,3 +9,17 @@
 @deprecate CategoricalVector(::Type{T}, m::Integer; ordered=false) where {T} CategoricalVector{T}(m, ordered=ordered)
 
 @deprecate CategoricalMatrix(::Type{T}, m::Int, n::Int; ordered=false) where {T} CategoricalMatrix{T}(m, n, ordered=ordered)
+
+function Base.convert(::Type{Nullable{S}}, x::CategoricalValue{Nullable}) where S
+    Base.depwarn("The Nullable type is deprecated. Use Missing instead.", :convert)
```
Comment (nalimilan) [inline src/deprecated.jl]: These methods only existed to fix ambiguities. I'd just put them under a `VERSION <=` check. We don't really care about `Nullable` in this package now.

File: test/runtests.jl (hunk 1 above)
Comment (ararslan) [reply test/runtests.jl]: I blame `sed`. :smile:

**Comment by ararslan** [comment]
Okay, this addressed a number of breakages and deprecations on 0.7, but I'm a little stumped on the `String` changes that are causing the failures now.

**Comment by ararslan** [comment]
@nalimilan Can we go ahead with this and address the string stuff separately?

**Comment by nalimilan** [comment]
Sure.

**Pull request merged**